### PR TITLE
Clean up runner

### PR DESCRIPTION
**Pull request by vdemedes**
Small cleanup of `lib/runner.js`

### CODE REVIEW

File: lib/runner.js [hunk 1]
```diff
@@ -207,17 +210,21 @@ Runner.prototype.run = function () {
...
+			}
 		})
 		.then(function () {
-			return eachSeries(tests.after, self._runTest.bind(self));
+			return eachSeries(tests.after, self._runTest, self);
 		})
 		.catch(noop)
 		.then(function () {
-			stats.testCount = tests.only.length ? tests.only.length : stats.testCount;
+			stats.testCount = hasOnlyTests ? tests.only.length : stats.testCount;
```
Comment (BarryThePenguin) [inline lib/runner.js]: What about...
```js
stats.testCount = tests.only.length || stats.testCount;
```

Comment (vdemedes) [inline lib/runner.js]: Good one!

Comment (BarryThePenguin) [inline lib/runner.js]: It's the small things :sparkles:

Comment (vdemedes) [inline lib/runner.js]: Fully agree!

**Comment by sindresorhus** [comment]
@jamestalmage Node.js 0.10 seems to be failing again on Windows :(

https://ci.appveyor.com/project/sindresorhus/ava/build/job/vrwhv5konnjgylcc

**Comment by jamestalmage** [comment]
Yes, failure is different now. "Missing plan".

In TAP, the plan line can be emitted at the beginning or end. If it is the end, we already know forked processes often drop a line or two of IO from the end on Windows (especially Node 0.10). It could be something else entirely, but that is my initial suspicion.

**Comment by sindresorhus** [comment]
@jamestalmage Yeah, my suspicion too. Might be relevant: https://github.com/nodejs/node/issues/3670

**Comment by jamestalmage** [comment]
Yes. Looks very relevant. Perhaps we need to avoid explicitly ending the process unless we have to.

**Pull request merged**